src/display/api.js

```diff
@@ -1302,11 +1302,25 @@ var PDFWorker = (function PDFWorkerClosure() {
           // https://bugzilla.mozilla.org/show_bug.cgi?id=683280
           var worker = new Worker(workerSrc);
           var messageHandler = new MessageHandler('main', 'worker', worker);
+          var terminateEarly = function() {
+            this._readyCapability.reject(new Error('Worker was destroyed'));
```

Comment:
Done; I've overwritten the previous commit, with the following changes (see PR for full change set).

```diff
@@ -1303,15 +1303,22 @@ var PDFWorker = (function PDFWorkerClosure() {
           var worker = new Worker(workerSrc);
           var messageHandler = new MessageHandler('main', 'worker', worker);
           var terminateEarly = function() {
-            this._readyCapability.reject(new Error('Worker was destroyed'));
-            messageHandler.destroy();
             worker.removeEventListener('error', onWorkerError);
+            messageHandler.destroy();
             worker.terminate();
+            if (this.destroyed) {
+              this._readyCapability.reject(new Error('Worker was destroyed'));
+            } else {
+              // Fall back to fake worker if the termination is caused by an
+              // error (e.g. NetworkError / SecurityError).
+              this._setupFakeWorker();
+            }
           }.bind(this);
 
           var onWorkerError = function(event) {
-            if (!event.message && !this.destroyed && !this._webWorker) {
-              // Worker failed to initialize due to a load error.
+            if (!this._webWorker) {
+              // Worker failed to initialize due to an error. Clean up and fall
+              // back to the fake worker.
               terminateEarly();
             }   
           }.bind(this);
```